src/XrdHttp/XrdHttpReq.hh

```diff
@@ -252,6 +252,9 @@ public:
 
   void addCgi(const std::string & key, const std::string & value);
 
+  // Check if the URL has a cgi parameter
+  static bool hasCGIParam(std::string_view url, std::string_view paramName);
```

Comment:
As per my [other](https://github.com/xrootd/xrootd/pull/2552/files#r2276593099) [comments](https://github.com/xrootd/xrootd/pull/2397#discussion_r2036767848), `std::string_view` is often misused, and it's used very little in XRootD currently. I'd strongly prefer to continue to use `const std::string&` instead and stay away from `string_view`. Sorry to be annoying about this, but I've had this position for a long time :-) See https://github.com/root-project/root/pull/2608#discussion_r218036622

@abh3 What's your opinion on this (i.e. `string_view` vs `const string&`)?

Reply:
I don't think we should change any major semantics or approaches as part of a bugfix.

(Especially as the proposals to make those changes requires breaking effectively all HTTP clients by default and making it so WLCG-specific clients are needed to use things.)

Reply:
Yes, however, @amadio argues that he wishes to do that as part of 5.9, and if we do that, we will not require the bug fix proposed in this PR.

Reply:
> I don't think we should change any major semantics or approaches as part of a bugfix.

That's not what we want here, but in XRootD 6.0, moving tokens to the query parameters of URLs will have to go away, as this is a security issue that needs to be addressed.

> Especially as the proposals to make those changes requires breaking effectively all HTTP clients by default and making it so WLCG-specific clients are needed to use things.

That's not true. Dropping `Authorization` happens [_by design_](https://datatracker.ietf.org/doc/html/rfc9110#section-15.4-6.3.1), for security reasons. In the future, we want to improve security by not bypassing this mechanism that is implemented in most clients. Using `curl --location-trusted` is a small price to pay for improving security.